federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package replicaset
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/golang/glog"
+
+	"fmt"
+	federationv1 "k8s.io/kubernetes/federation/apis/federation/v1beta1"
+	fedcache "k8s.io/kubernetes/federation/client/cache"
+	fedclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/meta"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	extensionsv1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	ExpectedReplicasAnnotation  = "kubernetes.io/target-cluster"
+	UserAgentName               = "Federation-replicaset-Controller"
+	CheckFedReplicaSetDelay     = time.Minute
+	localReplicaSetResyncPeriod = time.Minute
+)
+
+func encodeScheduleResult(map[string]int64) string {
+	return ""
+}
+func decodeScheduleResult(s string) (map[string]int64, error) {
+	return nil, nil
+}
+
+type ReplicaSetController struct {
+	// To allow injection of syncSubRC for testing.
+	syncHandler func(subRcKey string) error
+
+	fedClient    fedclientset.Interface
+	clusterCache *ClusterCache
+
+	clusterController *framework.Controller
+	clusterStore      fedcache.StoreToClusterLister
+
+	replicaSetController *framework.Controller
+	replicaSetStore      cache.StoreToReplicaSetLister
+
+	replicasetQueue workqueue.DelayingInterface
```

Comment:
I know, I'll adapt it to your deliverer once I sorted others. And the planner, too.